core/src/listener/state.ts

```diff
@@ -179,31 +175,25 @@ export class State {
     const contractAddress = toAddListener.address
     const observedBlockRedisKey = getObservedBlockRedisKey(contractAddress)
     const latestBlock = await this.latestBlockNumber()
+    const { blockKey: observedBlockKey } = await getObservedBlock({
+      blockKey: observedBlockRedisKey,
+      logger: this.logger
+    })
 
-    switch (this.listenerInitType) {
-      case 'clear':
-        // Clear metadata about previously observed blocks for a specific
-        // `contractAddress`.
-        await this.redisClient.set(observedBlockRedisKey, latestBlock - 1)
-        break
-
-      case 'latest':
-        await this.setObservedBlockNumberIfNotDefined(observedBlockRedisKey, latestBlock - 1)
-        break
-
-      default:
-        // [block number] initialization
-        await this.setObservedBlockNumberIfNotDefined(observedBlockRedisKey, latestBlock - 1)
-        for (let blockNumber = this.listenerInitType; blockNumber < latestBlock; ++blockNumber) {
-          const historyOutData: IHistoryListenerJob = {
-            contractAddress,
-            blockNumber
-          }
-          await this.historyListenerQueue.add('history', historyOutData, {
-            ...LISTENER_JOB_SETTINGS
-          })
-        }
-        break
+    /**
+     when listener starts, there are two options:
+      * latest observedBlock (key) exists -> do nothing 
+        (start latest job, it'll handle multiple blocks 
+        between observedBlock and latestBlock)
+      * it does not exist -> upsert latestBlock
+     */
```

Comment:
This is a bit peculiar comment formatting. How about this one?

```suggestion
    // When listener starts, there are two options:
    // 1) latest observedBlock (key) exists -> do nothing
    //    (start latest job, it'll handle multiple blocks
    //    between observedBlock and latestBlock)
    // 2) key does not exist -> upsert latestBlock
```